.github/workflows/ci.yml

```diff
@@ -660,10 +657,34 @@ jobs:
 
       - name: Coveralls Finished
         id: publish-coverage
-        uses: coverallsapp/github-action@a1a1a8a300a7e89df3630639df8fb23de5cc6368
+        uses: coverallsapp/github-action@v2
         with:
           github-token: ${{ secrets.GITHUB_TOKEN }}
           parallel-finished: true
 
       - name: Output Coveralls response
         run: echo ${{ steps.publish-coverage.outputs.coveralls-api-result }}
+
+  slack:
+    name: Slack
+    needs: 
+      - cleanup-dart-matrix
+      - cleanup-flutter-matrix
+    runs-on: ubuntu-latest
+    if: always()
```

Comment:
why don't we skip here if this is not main branch CI

Reply:
Let's leave it like this, On this way we make it sure that this step will run even though the previous `jobs` and `needs` fail.